File: ComputerSeekho.Net/ComputerSeekho.Net/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ComputerSeekho.Net.Models;

[Table("student")]
public class Student
{
    [Key]
    [Column("student_id")]
    [System.Text.Json.Serialization.JsonPropertyName("studentId")]
    public int StudentId { get; set; }

    [Column("payment_due")]
    [System.Text.Json.Serialization.JsonPropertyName("paymentDue")]
    public double? PaymentDue { get; set; }

    [Column("student_name")]
    [Required]
    [MaxLength(30)]
    [System.Text.Json.Serialization.JsonPropertyName("studentName")]
    public string StudentName { get; set; } = string.Empty;

    [Column("student_address")]
    [MaxLength(60)]
    [System.Text.Json.Serialization.JsonPropertyName("studentAddress")]
    public string? StudentAddress { get; set; }

    [Column("student_gender")]
    [MaxLength(10)]
    [System.Text.Json.Serialization.JsonPropertyName("studentGender")]
    public string? StudentGender { get; set; }

    [Column("photo_url")]
    [MaxLength(255)]
    [System.Text.Json.Serialization.JsonPropertyName("photoUrl")]
    public string? PhotoUrl { get; set; }

    [Column("student_dob")]
    [System.Text.Json.Serialization.JsonPropertyName("studentDob")]
    public DateOnly? StudentDob { get; set; }

    [Column("student_qualification")]
    [MaxLength(20)]
    [System.Text.Json.Serialization.JsonPropertyName("studentQualification")]
    public string? StudentQualification { get; set; }

    [Column("student_mobile")]
    [System.Text.Json.Serialization.JsonPropertyName("studentMobile")]
    public string? StudentMobile { get; set; }

    [Column("student_email")]
    [MaxLength(30)]
    [System.Text.Json.Serialization.JsonPropertyName("studentEmail")]
    public string? StudentEmail { get; set; }

    [Column("batch_id")]
    [System.Text.Json.Serialization.JsonIgnore]
    public int? BatchId { get; set; }

    [ForeignKey("BatchId")]
    [System.Text.Json.Serialization.JsonPropertyName("batch")]
    public virtual Batch? Batch { get; set; }

    [Column("course_id")]
    [System.Text.Json.Serialization.JsonIgnore]
    public int? CourseId { get; set; }

    [ForeignKey("CourseId")]
    [System.Text.Json.Serialization.JsonPropertyName("course")]
    public virtual Course? Course { get; set; }
}

File: ComputerSeekho/src/main/java/com/example/entities/Student.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "student")
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "student_id")
    private int studentId;

    @Column(name = "payment_due")
    private Double paymentDue;

    @Column(name = "student_name", length = 30, nullable = false)
    private String studentName;

    @Column(name = "student_address", length = 60)
    private String studentAddress;

    @Column(name = "student_gender", length = 10)
    private String studentGender;

    @Column(name = "photo_url", length = 255)
    private String photoUrl;

    @Column(name = "student_dob")
    private LocalDate studentDob;

    @Column(name = "student_qualification", length = 20)
    private String studentQualification;

    @Column(name = "student_mobile")
    private String studentMobile;

    @Column(name = "student_email", length = 30)
    private String studentEmail;

    @ManyToOne
    @JoinColumn(name = "batch_id", referencedColumnName = "batch_id")
    private Batch batch;

    @ManyToOne
    @JoinColumn(name = "course_id", referencedColumnName = "course_id")
    private Course course;

    public int getStudentId() {
        return studentId;
    }

    public void setStudentId(int studentId) {
        this.studentId = studentId;
    }

    public Double getPaymentDue() {
        return paymentDue;
    }

    public void setPaymentDue(Double paymentDue) {
        this.paymentDue = paymentDue;
    }

    public String getStudentName() {
        return studentName;
    }

    public void setStudentName(String studentName) {
        this.studentName = studentName;
    }

    public String getStudentAddress() {
        return studentAddress;
    }

    public void setStudentAddress(String studentAddress) {
        this.studentAddress = studentAddress;
    }

    public String getStudentGender() {
        return studentGender;
    }

    public void setStudentGender(String studentGender) {
        this.studentGender = studentGender;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public void setPhotoUrl(String photoUrl) {
        this.photoUrl = photoUrl;
    }

    public LocalDate getStudentDob() {
        return studentDob;
    }

    public void setStudentDob(LocalDate studentDob) {
        this.studentDob = studentDob;
    }

    public String getStudentQualification() {
        return studentQualification;
    }

    public void setStudentQualification(String studentQualification) {
        this.studentQualification = studentQualification;
    }

    public String getStudentMobile() {
        return studentMobile;
    }

    public void setStudentMobile(String studentMobile) {
        this.studentMobile = studentMobile;
    }

    public String getStudentEmail() {
        return studentEmail;
    }

    public void setStudentEmail(String studentEmail) {
        this.studentEmail = studentEmail;
    }

    public Batch getBatch() {
        return batch;
    }

    public void setBatch(Batch batch) {
        this.batch = batch;
    }

    public Course getCourse() {
        return course;
    }

    public void setCourse(Course course) {
        this.course = course;
    }
}